examples/io-descriptors/service.py

```diff
@@ -0,0 +1,108 @@
+import typing as t
+from pathlib import Path
+
+import numpy as np
+import pandas as pd
+import torch
+from PIL import Image as im
+from PIL.Image import Image
+from pydantic import Field
+
+import bentoml
+from bentoml.validators import DataframeSchema
+from bentoml.validators import DType
+
+# PROMPT_TEMPLATE = """<image>\nUSER: What's the content of the image?\nASSISTANT:"""
+
+
+@bentoml.service()
+class ImageResize:
+    @bentoml.api()
+    def generate(self, image: Image, height: str = "64", width: str = "64") -> Image:
+        size = int(height), int(width)
+        return image.resize(size, im.ANTIALIAS)
+
+    @bentoml.api()
+    def generate_with_path(
+        self,
+        image: t.Annotated[Path, bentoml.validators.ContentType("image/jpeg")],
+        height: str = "64",
+        width: str = "64",
+    ) -> Image:
+        size = int(height), int(width)
+        image = im.open(image)
+        return image.resize(size, im.ANTIALIAS)
+
+
+@bentoml.service()
+class AdditionService:
+    @bentoml.api()
+    def add(self, num1: float, num2: float) -> float:
+        return num1 + num2
+
+
+@bentoml.service()
+class AppendStringToFile:
+    @bentoml.api()
+    def append_string_to_eof(
+        self,
+        txt_file: t.Annotated[Path, bentoml.validators.ContentType("text/plain")],
+        input_string: str,
+    ) -> t.Annotated[Path, bentoml.validators.ContentType("text/plain")]:
+        with open(txt_file, "a") as file:
+            file.write(input_string)
```

Comment:
when writing a file, use a tmp directory like https://github.com/bentoml/BentoXTTS/blob/main/service.py#L37
same for audiospeedup example